[PlayStation] Avoid mincore() call in BlockDirectory::updatePercentageOfPagedOutPages()

unicodon opened the pull request:
#### 7fab05687df112cb8a846f4faef8c59ffae56bec
<pre>
[PlayStation] Avoid mincore() call in BlockDirectory::updatePercentageOfPagedOutPages()
<a href="https://bugs.webkit.org/show_bug.cgi?id=255202">https://bugs.webkit.org/show_bug.cgi?id=255202</a>

Reviewed by Ross Kirsling.

Until we figure out the best way on PlayStation to check if pages are paged out, disable
the code block of BlockDirectory::updatePercentageOfPagedOutPages() in the PlayStation port.

* Source/JavaScriptCore/heap/BlockDirectory.cpp:
(JSC::BlockDirectory::updatePercentageOfPagedOutPages):

Canonical link: <a href="https://commits.webkit.org/262766@main">https://commits.webkit.org/262766@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/03e7ec63891447aa5f5e877321029fe04be0cc78

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2558 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2561 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2684 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/3959 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/2952 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2702 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2648 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2260 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2582 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/2919 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2291 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/3724 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/503 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2276 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/2138 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/2137 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/2585 "Built successfully") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2298 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3490 "run-api-tests-without-change (failure)") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/2452 "Built successfully and passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2326 "Passed tests") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2108 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/2634 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2332 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2279 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/608 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/626 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2312 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/2690 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2463 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/723 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

rkirsling commented on Source/JavaScriptCore/heap/BlockDirectory.cpp:
```diff
@@ -59,8 +59,9 @@ void BlockDirectory::setSubspace(Subspace* subspace)
-#if OS(UNIX)
+#if OS(WINDOWS) || PLATFORM(PLAYSTATION)
+    // FIXME: We should figure out a solution for Windows and PlayStation.
+#elif OS(UNIX)
```
Even though we may want a common solution for Win & PS, I think it's probably best to just use `#if OS(UNIX) && !PLATFORM(PLAYSTATION)`?

unicodon replied:
Probably Win's solution will be different from PS's so I'll change that condition.

rkirsling reviewed (approved):
r=me

unicodon committed:
[PlayStation] Avoid mincore() call in BlockDirectory::updatePercentageOfPagedOutPages()
https://bugs.webkit.org/show_bug.cgi?id=255202

Reviewed by Ross Kirsling.

Until we figure out the best way on PlayStation to check if pages are paged out, disable
the code block of BlockDirectory::updatePercentageOfPagedOutPages() in the PlayStation port.

* Source/JavaScriptCore/heap/BlockDirectory.cpp:
(JSC::BlockDirectory::updatePercentageOfPagedOutPages):

Canonical link: https://commits.webkit.org/262766@main
